guides/notebooks/stylesheets.livemd

```diff
@@ -0,0 +1,456 @@
+
+SwiftUI employs **modifiers** to style and customize views. In SwiftUI syntax, each modifier is a function that can be chained onto the view they modify. Here's a basic example of making text red using the [foregroundStyle](https://developer.apple.com/documentation/swiftui/text/foregroundstyle(_:)) modifier:
+
+```swift
+Text("Some Red Text").foregroundStyle(.red)
+```
+
+Many modifiers can be applied to a view. Here's an example using [foregroundStyle](https://developer.apple.com/documentation/swiftui/text/foregroundstyle(_:)) and [frame](https://developer.apple.com/documentation/swiftui/view/frame(width:height:alignment:)).
+
+```swift
+Text("Some Red Text")
+            .foregroundStyle(.red)
+            .frame(height: 100, width: 100)
+```
+
+In LiveView Native, we use `~SHEET` sigil stylesheets to organize modifers by classes using an Elixir-oriented DSL similar to CSS for styling web elements.
+
+Here's the same `foregroundStyle` and `frame` modifiers grouped into a "color-red" class in a stylesheet:
+
+<!-- livebook:{"force_markdown":true} -->
+
+```elixir
+~SHEET"""
+  "color-red" do
+    foregroundColor(.red)
+    frame(height: 100, width: 100)
+  end
+"""
+```
+
+The DSL (Domain Specific Language) used in LiveView Native drops the `.` dot before each modifier, but otherwise remains largely the same. We do not document every modifier separately, since you can translate SwiftUI examples into the DSL syntax.
+
+There are instances where modifier argument values are not compatible with Elixir terms, but these are infrequent and beyond the scope of this guide. In most cases, you can write your LiveView Native templates in such a way that this is a non-issue.
+
+For more, you can find documentation and examples of modifiers on [Apple's SwiftUI documentation](https://developer.apple.com/documentation/swiftui) which is comprehensive and thorough, though it may feel unfamiliar at first for Elixir Developers when compared to [HexDocs](https://hexdocs.pm/).
+
+If you use Visual Studio Code, we strongly recommend you install the [LiveView Native Visual Studio Code Extension](https://github.com/liveview-native/liveview-native-vscode) which provides autocompletion and type information thus making modifiers significantly easier to write and lookup.
+
+## Writing Stylesheets
+
+This next section focuses on practical examples of building your own stylesheets as well as some useful patterns and general best-practices.
+
+<!-- livebook:{"break_markdown":true} -->
+
+### Stylesheet Modules
+
+Stylesheet modules define classes within the `~SHEET` sigil and must include `use LiveViewNative.Stylesheet, :swiftui`.
+
+In this example, the stylesheet generates a `color-red` class. This class applies the `[foregroundStyle](https://developer.apple.com/documentation/swiftui/view/foregroundstyle(_:)` SwiftUI modifier, which accepts an argument for color. In our example, we chose the `[.red](https://developer.apple.com/documentation/swiftui/color/red)` color.
```

Comment:
The `foregroundStyle` and `.red` links are not rendering correctly. Should it be written as:
```suggestion
In this example, the stylesheet generates a `color-red` class. This class applies the [`foregroundStyle`](https://developer.apple.com/documentation/swiftui/view/foregroundstyle(_:)) SwiftUI modifier, which accepts an argument for color. In our example, we chose the [`.red`](https://developer.apple.com/documentation/swiftui/color/red) color.
```